src/main/java/org/spongepowered/api/Engine.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api;
+
+import org.spongepowered.api.client.Client;
+import org.spongepowered.api.scheduler.Scheduler;
+
+/**
+ * Shared functionality between {@link Client} and {@link Server} engines.
+ */
+public interface Engine {
+
+    /**
+     * Gets the scheduler used to schedule tasks.
```

Comment:
What's special about this scheduler? Is it per "engine"? If there's a single player world, technically there's two engines present, no?

Reply:
Nothing special. This is a shared interface between `Client` and `Server`. Yes, when a client is running a singleplayer world it has two schedulers - one on `Client`, one on `Server.

Reply:
Can you make that clearer in the docs here?

Reply:
I don't see the need for further documentation here.

Even stating the fact that there may be two engines in SinglePlayer is implementation leak to the MCP implementation. Client itself is borderline that too but for the sake of the majority of our userbase, I deemed it okay to proceed.

Reply:
I would just like to have explanaton that there may be multiple engines, or temporary engines etc. because in the event that you have a plug-in on the client, and scheduling a task for the server, the task would need to be recreated during the server start up again later. This would make it clear that there’s the potential for multiple thread schedulers.